### DISCUSSION TITLE
GOODF - Manual referenced tremulants

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#40] [General]
Tremulants work as expected but noticed this:
![image](https://github.com/larspalo/GOODF/assets/6999990/f65a5c56-d4ae-464a-96a0-9f1561fe8dfa)

Why? What's the logic behind this?

**Comment by larspalo** [comment]
You tell me, it's you who designed the odf! No referencing is done automatically, it's you who have created them. I suggest you look at what windchests are affected by which tremulant and compare that to what stops/ranks/pipes are placed on respective windchest. The referencing of a tremulant to a manual is not for affecting the pipes of stops under that manual, it's for having the tremulants possible to add in divisionals for that manual.

To be clear: the manual itself doesn't decide what tremulant affect which pipe in which stop! Tremulants affects windchests and what pipes are ultimately placed on that windchest.

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/526431d6-2b03-4ea2-a82d-ff56e371ccba)

![image](https://github.com/larspalo/GOODF/assets/6999990/b906ade4-cc4f-4563-89fb-07548530bd44)

**Reply by Bumblebee001** [reply]
I did not select the tremulant from the manuals window.... I noticed the entries there after I had set the tremulants from the windchest window. They work fine - don't get me wrong but the entry as represented in the manual window looks odd!

**Reply by larspalo** [reply]
It shouldn't be possible that the tremulant references on the manual happen by themselves! Referencing the tremulants on the windchest, should not by itself add them to the manual.

But again, the only effect adding a tremulant to a manual is that it's possible to add that tremulant to a divisional! When that tremulant in fact is controlled by a switch it has no effect whatsoever - because then it's the switch that should be referenced on the manual to get it stored in the divisional. Test removing them from the manual and see for yourself.

**Reply by Bumblebee001** [reply]
I corrected the enteries for Tremulant and they still function OK.

Secondly, I followed what you did on your sample and extended to the rest of the stops. The Tutti does not play anything (silent) and the Reeds do not turn off. What could I be missing? I did create the additional T selectors and settings, and the additional entries for Tutti, Reeds off, Reeds off (NOT). Mystery!

**Reply by Bumblebee001** [reply]
PS I worked on the basis of your last sample. Until I hear from you I shall look back at the previous one you sent.

**Reply by Bumblebee001** [reply]
There were additional things in that and included them but made no difference

**Reply by larspalo** [reply]
Can't say what's different in your odf, as you sure can understand. Did you remember to add the T selector as activator in the (xor:ed) stops? My odf was working, right?

**Reply by Bumblebee001** [reply]
Sorted the tutti.....
Looking into the Reeds issue

**Reply by Bumblebee001** [reply]
I have two instances of GOODF running to compare step by step. I came to open the other sample you sent me and this popped up.... again! If I try to ignore or abort, things appear to get deleted from the open file.....!
![image](https://github.com/larspalo/GOODF/assets/6999990/3a286b1d-cbd5-4d78-a414-cf2d6cf2a639)

**Reply by Bumblebee001** [reply]
In your latest sample, the entry of tutti selector is stop specific. Which should I keep?
![image](https://github.com/larspalo/GOODF/assets/6999990/78f4771b-e7a6-4a4d-8ca8-9bb0cc3001e9)

**Reply by larspalo** [reply]
Exactly what are you aborting or cancelling? I fixed the crash happening from cancelling the open/load/read file operation earlier. For many items the open and cancel method is intended to clear the field (of a file that earlier was browsed for) - but it shouldn't cause a crash.

**Reply by larspalo** [reply]
The stop specific tutti selector is there to have the reeds off be effective in any other active combination.

**Comment by Bumblebee001** [comment]
I had your penultimate sample open and wanted to load the last one so I
clicked open to load the second file.

**Reply by larspalo** [reply]
Right, in 0.6.2 that would cause a crash, but on 0.6.3 it should work. Do you get a crash every time on 0.6.3?

**Comment by Bumblebee001** [comment]
I am using 0.6.3. First I get a message: all current data will be lost! do
you want to proceed? I never changed anything in this file!

I clicked yes but tried it again and behaved ok now!

**Reply by Bumblebee001** [reply]
Nope....... crashed again!

**Reply by larspalo** [reply]
Sorry, but I've been testing read/open with cancelling the operation in any stage (at warning dialog, at file choosing dialog), tested new organ with cancelling and it (seemingly) never crashes now on Linux as opposed to v0.6.2. I'll need the exact steps to reproduce it (and quite possibly a windows machine to test on, but I guess Wine will have to do).

**Comment by Bumblebee001** [comment]
Whiskey would be better🤣🤣🤣

**Reply by larspalo** [reply]
Agreed! Single Malt. Preferably from Islay.

**Comment by Bumblebee001** [comment]
I noticed something quite odd. I hope it's just me..... I reviewed my settings to match your reed stop 2 references. I had to bring the Reeds Off (NOT) up the list of switches so that I could select it for the stop selectors H, I, II, Tutti. I noticed that there was no reeds off  element yet the piston still appears in GO and in the layout screen. I created the element again and now I have 2 appearing on the screen and neither of them work! Crazy..... I think I've depleted all my brain neurotransmitters on this sample-set whilst briefing you every step of the way...... I'm drained!

**Reply by Bumblebee001** [reply]
PS I had to resort to NotePad++ and manually renumber the switches in between from the new place to where the entry was maintaining the numerical order. Both elements appear on NotePad++ but the GUI elements only shows one (the new one I created)

**Reply by larspalo** [reply]
Yes, it can be quite confusing. Just be careful if you re-number things in the odf to do it in all places, otherwise very strange things can happen. This will be much easier with the next release.

**Comment by Bumblebee001** [comment]
Oops I forgot that 🤯

**Reply by larspalo** [reply]
Been there, done that! Suddenly completely different references showed up where I expected others to be!

**Reply by Bumblebee001** [reply]
I think I succeeded in reverting everything back to what it was. I guess I shall have to wait for your update.

**Reply by larspalo** [reply]
Ok, I can trigger a new build right now, but I had originally much more in mind for the next release... More drag and drop options will be added, but already the four commits since last release is enough for changing version number since it introduces new features!

**Reply by Bumblebee001** [reply]
That would be nice and most helpful.

**Comment by Bumblebee001** [comment]
I am delighted by the improvements and the latest 0.7.0 is so far the best version yet. I have been able to drag a switch up as I needed and had to revise the entire odf largely because of my blunder in messing about the numerical order of switches in Notepad++ mindless of the references that would have needed to change accordingly. 

@larspalo 
I have finally got the Tutti and the Reeds off working. The samples you sent were indispensible in achieving what I have achieved and therefore many many thanks for your help in this regard too.

Now I have to tackle other features of this organ.

The GC piston currently shuts off all sound but it visually turns off the H tabs leaving I and II in the ON state (if they were on). I don;t know whether to, in some way, disable and visual changes or to extend the visual changes also to the I and II tabs. From what I understand, the GC should not effect the visible state (on/off) of anything and should only disable sound.

Then there will be the need to create switches and references for the third row of pedal stops and it should be possible to control their activation and deactivation by a piston - the Ped.Comb. piston.

There is also the crescendo pedal (the original organ has no "swell pedal") and this again is activated/ deactivated by a separate piston.

So there is some work still needs doing. I am working beyond my knowledge and limited expertise on the matter. I can do with all the help I can get on what's left to do.

**Reply by Bumblebee001** [reply]
I created these switches for the extra Pedal stops:
![image](https://github.com/larspalo/GOODF/assets/6999990/5878f1d1-18a6-4d52-9e99-71ad453ae498)

My primary query is whether I would need to add any of these to existing switches. These stops, when the Ped.Comb piston is activated, would over-ride whatever ped stops have been selected in whichever manual is active (H, I or II). I think so but this should be over-riding of pedal stops only not the entire manual.....

**Reply by larspalo** [reply]
You should create a III (and everything related to that) that will only affect the chosen pedal stops/couplers.

For the swell (crescendo), that's another matter... I'm, not finished with that yet!

**Reply by Bumblebee001** [reply]
I have added some references (to the new switches only) but I'm sure they are incomplete at best.

Furthermore, I added another swell pedal hoping this will serve as a crescendo pedal. Crescendo A only showed as a stop or piston not a pedal so I added this as the Cresc. Ab piston. For the swell, I typed Cresc as the name but "swell" appears NOT the name assigned.

**Reply by Bumblebee001** [reply]
I'll add III.... but I still need guidance to ensure the references are correct. I would understand I shall need to add appropriate references to this III in existing switches besides the Ped specific ones just added.

**Reply by Bumblebee001** [reply]
Would (Stop) III merely replace my current (Stop) P?

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/8058c8cc-34a5-48a1-8f20-f06dc2a37cd1)

**Reply by Bumblebee001** [reply]
I tried adding references but I noticed a number of issues.
1. Tutti now does not deactivate when other pistons are selected (H, I, II)
2. The Ped.Comb. piston comes off only when other pistons are activated (it does not go off when pressed a second time)
3. I cannot get the Ped stops to over-ride the selection on either of the 3 or 4 pistons (H, I, II, T)

.... and these are what I noticed..... there could be other issues I have not yet noticed. :-( (I am trying..... :'-(   )

**Reply by Bumblebee001** [reply]
I must have conflicting references...... Tutti Reeds are messed up now, when H, I, II are activated after tutti, organ goes silent. Ped stops don't play at all

**Reply by larspalo** [reply]
First of all one needs to decide on how things should behave. Since I don't have a clear picture of how the original organ exactly behaves it's a bit difficult for me to say exactly what you should do.

I'd guess that the third pedal combination when activated will "override" the H, I and II, when de-activated it might "fall back" to whatever of H, I or II is active. But I'm not sure about this. Since the "third" pedal combination only affects the pedal I think it's highly likely that the states of H, I or II will remain active but overridden for the pedal when the pedal combination button is active.

Also, a guess regarding the crescendo swell is that when it's activated it will "override" any other active combination (H, I, II or Ped) but not cancel them, but I'm not sure about this either. It's very likely that there won't be any indication on any of the stops for what step of the crescendo swell on is at - sometimes a kind of meter would show how far up or down on the crescendo one presently would be, but even this is not sure. From the wikipedia article on ["Crescendo pedal"](https://en.wikipedia.org/wiki/Crescendo_pedal) this comes:

> In actual use, the operation of the crescendo pedal usually does not move the draw knobs or stop tabs on the console; the stops are electronically activated inside the organ. Often an indicator light or lights will be present on the console to inform the organist of when the crescendo pedal is activated and how far it is engaged. 

This is quite problematic because of how the swell (crescendo) in GO is done and programmed (possible to program). My suggestion for you would be yet another layer of stop activating switches that would be enabled or disabled by the "Crescendo" button and only them should be "programmed" for the crescendo. The problem is that presently in GO there's no way to prevent the user from changing the crescendo programming and possibly do "wrong things" with it (like programming other switches to be activated by it than the ones you've decided). @oleg68 Perhaps it would be good idea to suggest it as an enhancement feature for GrandOrgue to have an odf crescendo that's not possible for the user to change (the A, B, C and D versions exist builtin in GO for four user programmable crescendos) to model such "fixed" register crescendos?

**Reply by Bumblebee001** [reply]
I have not been able to re-visit the organ yet. So I could not check things out but I believe what you have stated is all correct. I think we can go along with what you have described at least I have a good functional sample-set. I could warn users not to tamper with the odf unless they really know what they are doing - ideally, they should never do so with any set. There is always the possibility of them re-downloading the set from the given link and re-install.... unless I change the site where I would upload the sample-set.

There are two "meters" on the organ - one shows voltage/ power when the organ is on but I don't know what the other is meant to be. It might well be related to the activity/ level of the crescendo.

![WhatsApp Image 2023-04-25 at 16 34 12](https://github.com/larspalo/GOODF/assets/6999990/d3c32972-2373-45c9-96e5-a3a07ba25737)

![WhatsApp Image 2023-04-25 at 16 34 12 2](https://github.com/larspalo/GOODF/assets/6999990/bf134055-2140-4b06-961e-c6aa1a2abd06)

**Comment by Bombardon32** [comment]
Hello lars. 

1. I'd guess that the third pedal combination when activated will "override" the H, I and II, when de-activated it might "fall back" to whatever of H, I or II is active. But I'm not sure about this. Since the "third" pedal combination only affects the pedal I think it's highly likely that the states of H, I or II will remain active but overridden for the pedal when the pedal combination button is active.  *YES* correct

2. Also, a guess regarding the crescendo swell is that when it's activated it will "override" any other active combination (H, I, II or Ped) but not cancel them, but I'm not sure about this either. *YES* correct

Everything you have stated is correct Lars. That is exactly how it works.

**Reply by larspalo** [reply]
@Bombardon32 Thanks for confirming! I've read more about this kind of combination systems, and I even begin to suspect that the H is only "overrided" by the combinations I and II - not necessarily even visibly disabled (pushed out) and that the "Auslösung" would return the player to the H from any of the combinations previously activated when that button (Auslösung) is pushed. Do you know anything about that?

**Comment by Bumblebee001** [comment]
I am resisting making any additions or changes in my new sample-set using NotePad++ so that I can test GOODF at the various stages of sample-set construction. I hope that some features would not take too long to become available. I do appreciate that you are working hard at this and that you have a life beyond GOODF but I am very eager on testing out features as I go along and getting this sample-set released sooner rather than later.... and it's not that I do a lot of sample-sets to be able to test new features in the future! Currently I cannot proceed more than I have on my set and I am just waiting for developments from your end.

Regards
Mark

**Reply by larspalo** [reply]
> Currently I cannot proceed more than I have on my set and I am just waiting for developments from your end.

What kind of developments would be crucial to the sample set you're working on? The way I see it, an odf protected crescendo swell mode in GrandOrgue would be very beneficial, but it's likely going to take a long time before that happen... Most, if not all, other things could be done already - but I am willing to admit that this sample set of yours requires a switch network of considerable complexity to achieve having all features of the original organ present and working as they should.

There's not much in GOODF itself that can hold you back really - most things needed can be done already, but not always in the quickest or simplest of ways, I admit that too. There are many further improvements I have on the TODO list but none of them should be that important for you anyway.

**Comment by Bombardon32** [comment]
Hello Lars. I have already sent mark my spreadsheet regarding the Crescendo settings. 1 to 10 ect. 
![received_848122633325543](https://github.com/larspalo/GOODF/assets/140917877/2f487b3c-4f14-440e-897e-7d8a5c6a02a3)

**Comment by Bumblebee001** [comment]
I understand. I think you know just as much about the sample set and what
needs to be achieved. Anything that can help would be appreciated with
great anticipation. Crescendo pedal for a start. I have of chart of each
step and what stops should be on with help from Aaron and another well
known organist. Other things are already achievable but I’m getting
references screwed up ….

**Reply by Bumblebee001** [reply]
Aaron beat me to it. I can assign these details but I need to have a functional option on GOODF.

A Crescendo pedal is a common feature on organs and should be a priority for many if not most users of your software, I believe.

**Comment by Bombardon32** [comment]
That's understandable ☺️

**Reply by Bumblebee001** [reply]
:-P

**Reply by larspalo** [reply]
Ok, so the crescendo pedal itself is a "Setter element" called "Swell". (Ignore most of the other related buttons for now, except possibly the meter that could maybe be the "CrescendoLabel" and if you do a separate panel like I suggest below you might want Set, CrescendoNext, CrescendoPrev and CrescendoCurrent too on that panel to make the programming a lot easier).

The actual programming of it is almost the last thing you do with your sample set before distribution. It's done by opening a final odf file in GrandOrgue (preferably completely clean, with no other MIDI connections that may be for your own use) and program the crescendo swell with each of the steps in the spreadsheet (use the set button and move one step at a time and select the switches that should be active at that step to store them).

For the crescendo you should have another set of switches that will allow to activate the stops only when the Crescendo is selected to be active. Personally, I would have created a separate panel that only display the switches that are supposed to be programmed into the crescendo together with the buttons that are helpful for doing the actual programming (which is not the same as H, I, II or Ped comb). On the main panel you only show the crescendo swell pedal and the button to activate/de-activate the crescendo.

Finally, when everything works as expected, you export the settings from GrandOrgue to a .cmb file name with the _same name_ as your .organ file and place it in the _same directory_. What this will do is that GO will automatically import the settings and apply them as defaults when the user load/open the sample set.

You must understand that since there are no odf related options to the setter element of a crescendo swell at the moment, nothing beyond creating/displaying the elements (and a related panel for easier programming) can be done in GOODF. All the rest must be done in GrandOrgue for the moment being.

**Reply by larspalo** [reply]
Actually, thinking of it, I believe that for the main panel you should only use the "Swell" setter element and not the CrescendoLabel (that you should use on the separate panel instead). For the actual crescendo meter on the main panel you should just duplicate the "Swell" element with a different set of bitmaps showing the needle position changing as the swell pedal position is changed.

**Reply by Bumblebee001** [reply]
On the real organ, there is no "swell" pedal; just a crescendo!

**Comment by Bombardon32** [comment]
Exactly what I thought Lars. That does make perfect sense.

**Comment by Bumblebee001** [comment]
I'm having a problem with the Tutti.
Is this correct:
![image](https://github.com/larspalo/GOODF/assets/6999990/ea392eef-a7af-4aa8-8ecb-5d4cb542be37)

**Reply by larspalo** [reply]
I don't know if you implemented it like I did. But I had only the input for the general.

**Reply by Bumblebee001** [reply]
I am not understanding fully the meaning and effect of the various selectors, inputs etc so i am trying to copy what you did. I have 4 examples that you sent me and they do not tally in some respects so I don't know which I am supposed to follow. "Merging" two different references seems to result in undesired effects.... eg Tutti does not go off when I select other pistons or tutti comes on when the pedals stops are selected..... (PS I am just testing pedal stops not the other manual stops at present but I have looked at other stops to check conformity in references)

**Reply by larspalo** [reply]
My terminology is almost arbitrary, pay no greater heed to it... Though, it seems that you're not getting the principles that I've tried to convey in the examples, right. Again as I told in https://github.com/larspalo/GOODF/discussions/35#discussioncomment-6520846 it's pretty difficult to explain in words what needs to be done, which is why I've given you images and practical (very confined and limited) examples instead.

Bottom line, though, is that you cannot have a single direct line of activation of a "Stop" from (only) one tab in this system. Switches are used to control the path of on/off signals that ultimately will depend on the on/off state of many buttons/tabs/switches. I admit that some of the choices I've made could be done in different ways and still get the job done, I just made my choice from what appeared most logical to me and went with it. To overcome the "feedback" limitation of Switches in GrandOrgue I used on occasion protected Generals to activate/de-activate appropriate switches at once. Exactly the same mechanism is not needed for every type of feature of the organ - nor is it even possible, as some of them should have higher precedence than others and some of them are supposed to be possible to be used "on top" of each other.

The "Stops" have the function XOR so that one and only one other signal source can activate them, if two sources for some reason would be active at the same time - the stop will not be active. They are activated by different (Selector) switches depending on what signal path is currently the one that should be active. (Should the stop state be decided from the H, I, II or Ped, Tutti and eventually the crescendo)

The "Selectors" have the function AND, which means that all their listed switches must be true (or on) for them to be activated.

There are "Inputs" for switches that are actually not shown and in many cases only receive the signals from the "Generals". The user is not supposed to interact directly with the input switches but instead going through the "Generals" that can simultaneously send on/off signals to many elements. I have called them inputs, or activators in my first image schematic.

There are "State" switches that _show_ the state of the selected signal path. In the real organ this is also receiving interaction from the user, but in the model they only display their state as they must be possible to control from some of the other buttons, while still other buttons may decide that some of the stops may or may not sound (ie. Reeds Off). The state switches while being shown must have their mouse rectangle minimal to allow the user to interact with the transparent General that is layered on top of the state switch.

The state switches are many times also reversed by a complimentary NOT switch that's there to be able to disable other switches when another path for the signal is chosen.

There you have the basics explained in words as good as I can without writing a book on the subject. I'd recommend that you experiment with one of the free logical gate simulators that exist on the web to get a grip on how the signal path will/can change depending on the state of the different switches along the path.

**Reply by larspalo** [reply]
I have for instance tested https://academo.org/demos/logic-gate-simulator/ that is somewhat limited compared to what the switches in GrandOrgue can do, but it's a nice help to understand the signal path flow. Think about the Input as the points where the user interacts and the Output is the Stop active on/off.

**Reply by Bumblebee001** [reply]
When I was young, I had trouble with mathematics. What was logical to most was an enigma to me. I tried hard to understand things which I eventually did and got the highest grade in my school leaving exams (Oxford University if you please!). It took time then but at 61 I am not that hopeful I can figure the logic with several switches interacting with one another with arguement like "if", "and", "xor", "not" etc. Can you imagine me trying this with MS Excel's formulae? When I need to do a spreadsheet with a lot of: if this, then that, but not if the other..... then I was in the middle of the Pacific ocean struggling against the waves in the middle of a moonless night completely disoriented and helpless..... and clueless to where I was!

You have explained it very eloquently, I must admit and your initial diagram was a bit of a respite but I cannot say I am going to troubleshoot successfully and find why something doesn't do what I expect it to do. At present, I select T and the pedals all go off! And I'm sure I followed your examples..... but there are so many entries now, I would not be surprised to have missed something out altogether (like selecting that "selected switch is on vs off" for a particular reference........)

**Reply by Bumblebee001** [reply]
That link is very interesting and content worth remembering.

**Comment by Bumblebee001** [comment]
Is this entry correct:
![image](https://github.com/larspalo/GOODF/assets/6999990/71e3e8c7-35c9-4240-82aa-690964aea803)

**Reply by larspalo** [reply]
That would depend on if you want the Tutti to override the Pedal Combination or not. Decisions... What should/could be more important? I'm not sure how the actual organ would behave, but I'd almost believe that the Tutti would override the pedal combination. In my examples I have actually also shown how to exclude one of the stops from the Tutti - and used the same stop to show how to add the extra (pedal) combination. Since I didn't know exactly how the choice should be made from the original organ, I left it up to you to decide.

**Reply by Bumblebee001** [reply]
Tutti should override everything by it's very nature!

**Reply by Bumblebee001** [reply]
As things are at present, tutti does not play any pedal stop. I must have something missing!

**Reply by larspalo** [reply]
It could be that Tutti would override and disable the Reeds Off, but I'm not completely certain about it. I can very well imagine that the organ builder, instead of providing a separate Plenum combination could have just made the Tutti and Reeds Off possible to combine. To be sure, one would need to be at the instrument and test it. In GO it's possible to make it behave any way you like it.

**Reply by Bumblebee001** [reply]
Reeds off are just for 2 stops that tutti should NOT activate (Gamba 8 and Vox Humane).... According to Aaron, these two stops, however, do become activated in the crescendo (though I do have my doubts from what the Maltese organ builder stated. These two stops "do not add anything meaningful to the Tutti and, if anything, have a negative impact..." So why should they ever come on with the crescendo?

**Comment by Bombardon32** [comment]
Right gentlemen, The Tutti does override the pedal pistons. It overrides
the whole instrument tbh like the Crescendo.

**Reply by larspalo** [reply]
Right, but if any (other) combination is pushed when the Tutti is active - the Tutti is then de-activated and the pushed combination goes active, right? If "Reeds Off" is pushed when Tutti is active, what happens? Does "Reeds off" stay active and in effect when another combination is activated or is it de-activated at any change? What is the purpose and function of the "Auslöser" button? What exactly happens to the state of the Handreg. button when Comb I or Comb II is pushed? The crescendo swell needs to be de-activated to not override everything else anymore (sounded like that in an earlier comment), right? We need to know exactly what each button does in relation to the others, which interact and how etc...

**Comment by Bombardon32** [comment]
Also Tutti does not active Gamba or Vox Humana or tremulant.

**Reply by Bumblebee001** [reply]
Step 10 of the crescendo should be Tutti!

**Comment by Bombardon32** [comment]
Correct

**Comment by Bumblebee001** [comment]
Still cannot figure out why the pedal stops won't come on when the tutti is pressed. I tried all sorts of combinations.

**Reply by Bumblebee001** [reply]
I'm taking a break of a few hours as I need it!

**Reply by larspalo** [reply]
Do so, then check the stops that they have the tutti selector listed there.

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/7bb51bb7-486e-46bb-b717-964dd4f48143)

**Reply by Bumblebee001** [reply]
Ok.... I think I found the issue: StopName P ...... state when activated currently "off"! Others are "no change" (not that I do understand the difference.... but hey ho!)

**Reply by Bumblebee001** [reply]
Nope....... no difference!!!

**Reply by Bumblebee001** [reply]
No wonder..... It is the state induced by the GC piston..... so it would not have any effect on the issue at hand.

**Comment by Bumblebee001** [comment]
Pistons seem to function as expected. The only issue at present is that
tutti activates everything except the pedal stops!

**Reply by larspalo** [reply]
If all the stops of the pedal has the tutti selector, then the next step is to check the Tutti selectors so that they have the correct switches. Finally there should only be the Tutti input that itself is controlled from the Tutti General.

What can happen too, since the stops are XOR, is that if some other selector still is active for the pedal stops, then they won't activate. You really need to make sure all other selectors for the pedal stops will be off when the Tutti is activated.

**Reply by Bumblebee001** [reply]
In your example, Stop 3 has no T Selector. Is that correct?

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/75f6cc59-712e-4337-a3c6-c00b1b8f823b)

**Reply by Bumblebee001** [reply]
Probably because this is related to the third row of stops unique to the pedals..... and have no place in the tutti selection.

**Reply by larspalo** [reply]
It's just to show you how you can exclude a stop from the Tutti by not including it like for the Stop 3. Also I just added the Stop 3 (Ped selector) to show how you add the third combination only to some of the stops. In both cases it could have been any of the stops really...

**Reply by Bumblebee001** [reply]
Yes.... got that..... it's keeping some things in mind whilst working along.... Now back to trying to troubleshoot those pedals stops..... I have found a tutti input for pedal stops but not for the rest of the stops yet the rest of the stops seem to come on but not the pedals..... These are the things that confuse me further.

**Reply by larspalo** [reply]
You should really only need one single Tutti (input) that all "Stop X (T selectors)" should reference if the stops should be affected by the Tutti.

**Reply by Bumblebee001** [reply]
I can send you the odf I created. I am no where near solving this issue and I have almost given up. I don't know what else to look at and check.

**Reply by larspalo** [reply]
Unfortunately I'll need the whole sample set to be able to open it in GOODF, or you need to change all the pipes to DUMMY ones.

**Reply by Bumblebee001** [reply]
I think I have located the error. Whilst I did include the "Tutti selector", I did not included the "STOPNAME T selector" in the pedal stops and couplers referenced list!

**Reply by larspalo** [reply]
Great! But check your generals too, as I mentioned in the other post. You only need the (Input) switches for the Tutti "General".

**Reply by Bumblebee001** [reply]
I think I understood what you meant. Now how can I easily remove all the 114 entries I incorrectly added? No remove all button available!

**Reply by larspalo** [reply]
> No remove all button available!

Well, yes, that could indeed be a problem... I might add the possibility to multi select in more or less all the listboxes when I have time to implement it!

**Reply by Bumblebee001** [reply]
Done. In this sampleset however it seems like the saved "stops" or switches in this case do not flow from one general to the next. When moving from 1 to 2 or some other of the 5 generals, the sound shuts off until I press the Hand Reg piston. Should the off be selected instead of on? I'll try it out tomorrow as I have had enough today! :-D

**Reply by Bumblebee001** [reply]
OK that was it!
Solved the Tutti issue and removed the generals altogether and re-did them.... but the issue mentioned in my last post remains.

**Reply by larspalo** [reply]
You do realize that the other generals you have there will need to match exactly one of the stop activating options only, and they will only work within that set of tabs - it's more than likely just even more confusing considering the type of combination system the organ in question have. Personally, I would have dropped the other generals as they don't belong to the original organ anyway.

**Comment by Bombardon32** [comment]
Very confusing.com however it looks like items are interfering with other
items so it's conflicting the settings from what I can gather.

**Comment by Bombardon32** [comment]
Please remove my last comment. I was being to quick to judge on matters. I
need to understand how the software is working and how the ODF is being
written first. Apologies Lars and your doing a fantastic job btw.

**Reply by Bumblebee001** [reply]
You can do that yourself by clicking on the 3 dots on the top right. We cannot do it ourselves as the delete option will not appear to us but only to you as the author of the posting.

**Reply by Bumblebee001** [reply]
![image](https://github.com/larspalo/GOODF/assets/6999990/afbd92e0-9470-421d-b363-56b093b2b394)

**Comment by Bumblebee001** [comment]
Lars, since the stops on my sample-set are **_switches_**, would creating the crescendo through GO be the same as when the stops are stops?

**Reply by larspalo** [reply]
The stops are stops but the switches are drawstops... Understand that as soon as an object has another function than Input - that should generally not be programmed into a combination as it's not activated by itself anymore - it's controlled by another element and in turn, that element is the one - which state should be saved into a combination of any type.

For this type of organ combination system there should be a special set of activating switches setup for the Crescendo Swell and only them should be programmed into the swell so that the swell can be activated or de-activated at will. This will basically be acomplished in a similar way as the H, I or II - but separate of course, and it should only be displayed on a special swell programming panel as you must have a way to set it up. (When the .cmb is exported it should be possible to remove the programming panel, but it's nothing I've tested)

**Comment by Bumblebee001** [comment]
As I have explained before the combined features I have will not
necessarily appear together in the final release. I also understand that a
simplified sample set will be using stops not switches and eliminating the
complex tab system altogether. I am just playing around with features and
options. Final decisions will be taken when I have addressed all the
issues. The only pending one at present is the crescendo.

**Reply by Bumblebee001** [reply]
How do I get the Crescendo Ab piston to work?

**Comment by Bumblebee001** [comment]
Also: I ran odfedit on my .organ file and I get this report:

ODF data check report :
WARNING in Panel000Element003 : the attribute DispImageNum is not expected in this object section or is misspelled
WARNING in Panel000Element004 : the attribute DispImageNum is not expected in this object section or is misspelled
WARNING in Panel000Element005 : the attribute DispImageNum is not expected in this object section or is misspelled
WARNING in Switch123 : the attribute Switch001 is not expected in this object section or is misspelled
WARNING in Switch124 : the attribute Switch001 is not expected in this object section or is misspelled
WARNING in Switch125 : the attribute Switch001 is not expected in this object section or is misspelled
WARNING in Switch126 : the attribute Switch001 is not expected in this object section or is misspelled
WARNING in Switch128 : the attribute Switch001 is not expected in this object section or is misspelled
WARNING in Switch130 : the attribute Switch001 is not expected in this object section or is misspelled
10,399 attributes checked
New panel format

What should I make out of this?

**Reply by larspalo** [reply]
Let me guess for the first three warnings:

Panel000Element003 to 005 are manuals using the new(er) style of built in bitmaps, right? It's likely that OdfEdit is not up to date with the more recent (3.12+) changes that has happened in GrandOrgue, so it doesn't recognize that DispImageNum for manuals is a valid entry now in the odf.

For the switches I cannot be sure without examining the relevant and related .organ file sections. I could imagine that if the Function of the switch (123, 124 etc) is NOT (which can only have one switch reference) then OdfEdit doesn't know that the line SwitchCount is not needed in the odf (and thus GOODF won't write it as GO will understand it anyway and even explicitly have noted the behaviour in the help) but OdfEdit still expects it.

TBH, worry about the warnings or errors that pop up when you open/load an organ in GrandOrgue! The OdfEdit warnings could just be caused by its programmers lack of understanding of the odf format. In _my opinion_ OdfEdit is a great tool (or rather has the potential to become one) for converting HW sample sets to GO, but beyond that there is little I would use it for...
